src/whisk.rs

```diff
@@ -293,7 +295,9 @@ pub fn rand_scalar<T: RngCore>(rng: &mut T) -> Fr {
 /// Serialize field element to bytes
 pub fn to_bytes_fr(fr: &Fr) -> FieldElementBytes {
     let mut bytes = [0u8; FIELD_ELEMENT_SIZE];
-    fr.write(&mut bytes[..]).unwrap();
```

Comment:
These write methods were a bit dangerous/ambiguous in arkworks <0.4 which is why there were removed. I'd recommend redoing the serialization strategy and testing it thoroughly -- if you are only using arkworks you can use the serialization as is, but if you are trying to interop with other libraries/languages, they will need to look at how arkworks does it and if it changes in arkworks then its a bit of a pain to see why and how.

Reply:
Is serialization of scalars spec'ed or standarized outside of Ethereum's specific context?

Reply:
There's usually a "standard-by-consensus" way to serialize a scalar per curve (or else my implementation of the curve will not interoperate with yours), but it's generally not spec'd. Some libraries will parametrise the endianness allowing the user to serialise using whichever endian they deem fit